tests/test_constrain_current.py

```diff
@@ -2,118 +2,190 @@
+                for dr, name in enumerate(names)
             }
-            transforms = get_transforms("iota_rr", eq=eq, grid=grid)
-            profiles = {"iota": None, "current": _ExactValueProfile(eq, grid)}
-            # compute rotational transform from the above current profile, which
-            # is monkey patched to return a surface average of B_theta in amps
-            data = compute_fun(
-                ["iota", "iota_r", "iota_rr"],
-                params=params,
+            # Compute rotational transform using the below toroidal current,
+            # which is monkey patched to return equation 6 in amps.
+            monkey_patched_profile = _ExactValueProfile(
+                eq=eq, grid=grid, profile_name="current"
+            )
+            actual = compute_fun(
+                names=names,
+                params=dict(params, c_l=None, i_l=None),
                 transforms=transforms,
-                profiles=profiles,
+                profiles=dict(profiles, current=monkey_patched_profile, iota=None),
             )
-            # compute rotational transform using the equilibrium's default
-            # profile (directly from the power series which defines iota
-            # if the equilibrium fixes iota)
-            benchmark_data = compute_fun(
-                ["iota", "iota_r", "iota_rr"],
-                params=get_params("iota_rr", eq=eq, has_axis=grid.axis.size),
+
+            # Evaluating the order n derivative of the rotational transform at
+            # the magnetic axis using a given toroidal current profile requires
+            # the order n + 2 derivative of that toroidal current profile.
+            # Computing the order n derivative of the resulting toroidal current
+            # as a function of rotational transform requires the order n
+            # derivative of the rotational transform.
+            # Therefore, this test cannot be used to cross-validate the magnetic
+            # axis limit of the two highest order derivatives of the rotational
+            # transform in data_index.
+            at_axis = grid.nodes[:, 0] == 0
+            for index, name in enumerate(names):
+                # don't try to validate if we can't compute axis limit yet
+                validate_axis = (
+                    index < len(names) - 2
+                ) and name not in not_implemented_limits
+                mask = (validate_axis & at_axis) | ~at_axis
+                np.testing.assert_allclose(
+                    actual=actual[name][mask],
+                    desired=desired[name][mask],
+                    equal_nan=False,
+                    err_msg=name,
+                )
+
+        # Only makes sense to test on configurations with fixed iota profiles.
+        test(DSHAPE, QuadratureGrid)
+        test(DSHAPE, ConcentricGrid)
+        test(DSHAPE, LinearGrid)
+
+    @pytest.mark.regression
+    @pytest.mark.solve
+    def test_current_to_iota_and_back(self, HELIOTRON_vac):
```

Comment:
> I think this should be HELIOTRON_vac2 which is the fixed current one

No, both `HELIOTRON_vac2` and `HELIOTRON_vac` have current profiles assigned and lack rotational transform profiles. Solving the `HELIOTRON_vac` input file (on `master` and `add_all_limits`) outputs an equilibrium where `profiles["iota"]` is not assigned and is `None` and `profiles["current"]` is some `PowerSeriesProfile`. 


> Not sure what you mean here? HELIOTRON_vac was still a "fixed iota" case, in that it didn't have a current profile assigned, so this test should work correctly.

The test `test_iota_to_current_and_back` only makes sense on equilibrium that know the rotational transform iota analytically. So any equilbrium used for this test should have something like a power series assigned to the rotational transform profile. Since `HELIOTRON_vac` did not have an assigned an iota profile, calling `eq.compute("iota")` on it will compute the rotational transform through equation 18 (as a function of toroidal current). This is not desireable because that's what we are testing. We want to know if that computation gives the same result as some power series that we knew beforehand.

You can confirm that `HELIOTRON_vac` does not work for this test, because calling `profiles["iota"].compute` on it, as done in this PR (see https://github.com/PlasmaControl/DESC/blob/e7d8c85be07361bd7e5d90f70bcdd970a2adac69/tests/test_constrain_current.py#L79) generates a None type error since `profiles["iota"]` is not assigned.

Reply:
> Could also use DSHAPE_current
I remember when this test was originally written Daniel wanted me to test this with a stellarator too.

Reply:
ah ok i forgot that we changed the way we do vacuum stuff. We should probably combine the two heliotron vac fixtures and check that we're using them in the right place (will make another PR)